homeassistant/components/cloud/backup.py

```diff
@@ -109,63 +104,14 @@ async def async_download_backup(
             raise BackupAgentError("Backup not found")
 
         try:
-            details = await async_files_download_details(
-                self._cloud,
+            content = await self._cloud.files.download(
                 storage_type=_STORAGE_BACKUP,
                 filename=self._get_backup_filename(),
             )
-        except (ClientError, CloudError) as err:
-            raise BackupAgentError("Failed to get download details") from err
-
-        try:
-            resp = await self._cloud.websession.get(
-                details["url"],
-                timeout=ClientTimeout(connect=10.0, total=43200.0),  # 43200s == 12h
-            )
-
-            resp.raise_for_status()
-        except ClientError as err:
+        except CloudError as err:
             raise BackupAgentError("Failed to download backup") from err
```

Comment:
Is there a message attached to the exception caught that we want to propagate and attach to the exception we raise?

Reply:
Not yet; there is a followup PR planned with translated errors for this.

Reply:
Eg:
```suggestion
            raise BackupAgentError(f"Failed to download backup: {err}") from err
```